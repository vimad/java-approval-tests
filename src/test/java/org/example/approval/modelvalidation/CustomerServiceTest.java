package org.example.approval.modelvalidation;

import org.approvaltests.Approvals;
import org.example.approval.model.Customer;
import org.example.approval.service.CustomerService;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class CustomerServiceTest {

    private final CustomerService customerService = new CustomerService();

    @Test
    public void traditionTesting() {
        Customer customer = customerService.createCustomer();

        assertEquals(1L, customer.getId());
        assertEquals(30, customer.getAge());
        assertEquals("vinod", customer.getFirstName());
        assertEquals("madubashana", customer.getLastName());
        assertEquals("test@gmail.com", customer.getEmail());
        assertEquals("Sri Lanka", customer.getCountry());
        assertEquals("address", customer.getAddress());
        assertEquals("vinod", customer.getUserName());
        assertTrue(customer.isActive());
    }

    @Test
    public void approvalTestingUsingTextRepresentation() {
        Customer customer = customerService.createCustomer();

        Approvals.verify(customer);
    }

}
