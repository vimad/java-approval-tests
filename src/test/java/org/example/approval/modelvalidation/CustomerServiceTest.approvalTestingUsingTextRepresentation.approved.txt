Customer(id=1, age=30, firstName=vinod, lastName=madubashana, country=Sri Lanka, email=test@gmail.com, userName=vinod, address=address, isActive=true)